esp/templates/survey/survey_form.html

```diff
@@ -1,125 +1,56 @@
-                    </td>
-                    <td>
-                        <select name="attendance_{{ ts.id }}" id="id_attendance_{{ ts.id }}">
-                            <option value="0">[None]</option>
-                            {% for sec in ts.classsections %}
-                                <option value="{{ sec.id }}" {% if sec.selected %}selected="selected"{% endif %}>{{ sec.parent_class.title }}</option>
-                            {% endfor %}
-                        </select>
-                    </td>
-                </tr>
-                </table>
-                <div id="perclass_questions_{{ ts.id }}">
-                {% for question in perclass_questions %}
-                    {{ question.render|substitute:ts }}
-                {% endfor %}
-                </div>
-            {% else %}
-                <table class="fullwidth">
-                <tr><td style="text-align: center;">You have already filled out a survey for a section at this time.</td></tr>
-                </tr>
-                </table>
-            {% endif %}
-        {% endfor %}
-    {% else %}
+{% if view_list %}
+    <p class="message">
+    {% if submitted or completed %}
+    <span style="color: red; font-weight: bold;">
+    {% if submitted %}Your survey responses for {% if section %}{{ section }}{% elif general %}the general program survey{% endif %} have been saved.
+    {% elif completed %}You've already completed the {% if section %}survey for {{ section }}{% elif general %}general program survey{% endif %}.{% endif %}
+    Please select another survey component below or come back later.
+    </span>
+    <br /><br />
+    {% endif %}
+    This survey consists of a survey component for each class you are registered for and a general program survey component. You may fill out a class survey once the class has begun and you may fill out the general program survey once your first class has begun. If no survey components are available, please come back later.
+    </p>
+    <table class="fullwidth">
+        <tr>
+            <th>Survey Component</th>
+            <th>Status</th>
+        </tr>
         {% for sec in sections %}
-          <table class="fullwidth">
-           <tr>
-                <th>Questions for {{ sec.emailcode }}: {{ sec.title }} ({% for tm in sec.friendly_times %}{% if forloop.counter > 1 %},{% endif %}{{ tm }}{% endfor %})</th>
-            </tr>
-            </table>
-            {% for question in perclass_questions %}
-                {{ question.render|substitute:sec }}
-            {% endfor %}
+        <tr>
+            <td class="underline" align="center">{{ sec }}</td>
+            <td class="underline" align="center">{% if not sec.started %}<span style="color: grey">Not available</span>{% elif sec.completed %}<span style="color: green">Completed</span>{% else %}<a class="btn fancybutton no-dec" href="{{ request.path }}?sec={{ sec.id }}">Take Survey</a>{% endif %}</td>
```

Comment:
This worked time-wise, but you can bypass the not available by going through the section details page and clicking the survey link there. Could you also put a "not available until class starts" message instead of the link on the survey details page?